### PR TITLE
node/express task part 2

### PR DESCRIPTION
**Pull request by dzmitrybek**

### CODE REVIEW

File: src/app/AppView.js [hunk 1]
```diff
@@ -198,21 +260,33 @@ export default class AppView {
...
-                callback(+event.target.getAttribute('data-id'));
+                callback(event.target.getAttribute('data-id'));
             }
         });
     }
 
-    onMyNewsPage(callback) {
-        this.myNewsPageBtn.addEventListener('click', () => {
+    onUserPageBtn(callback) {
+        this.userPageBtn.addEventListener('click', () => {
```
Comment (slowpokex) [inline src/app/AppView.js]:
```
 this.userPageBtn.addEventListener('click', callback);
```
Maybe like as above?

File: src/app/AppView.js [hunk 2]
```diff
@@ -198,21 +260,33 @@ export default class AppView {
...
+            callback();
+        });
+    }
+
+    onSignInPageBtn(callback) {
+        this.signInPageBtn.addEventListener('click', () => {
+            callback();
```
Comment (slowpokex) [inline src/app/AppView.js]: The same

File: src/server/models/news/model.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+const mongoose = require('mongoose');
+
+const Schema = mongoose.Schema;
+
+const newsSchema = new Schema({
+    _id: {
```
Comment (slowpokex) [inline src/server/models/news/model.js]: You can omit declaring this property

File: src/server/models/user/model.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
+const mongoose = require('mongoose');
+
+const Schema = mongoose.Schema;
+
+const userSchema = new Schema({
+    _id: {
```
Comment (slowpokex) [inline src/server/models/user/model.js]: SAme you can omit

File: src/server/models/user/router.js [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    );
+
+router.route('/api/logout')
+    .get((req, res) => {
+        req.logout();
+        res.sendStatus(200);
+    });
+
+router.route('/api/registration')
+    .post(async (req, res, next) => {
```
Comment (slowpokex) [inline src/server/models/user/router.js]: async is redundant because you have used promise without await

File: src/server/server.js [hunk 6]
```diff
@@ -1,87 +1,39 @@
...
-const fs = require('fs');
-const util = require('util');
+const config = require('./config');
 
-const readFileAsync = util.promisify(fs.readFile);
-const writeFileAsync = util.promisify(fs.writeFile);
+mongoose.Promise = Promise;
 
-const storageFilePath = path.resolve(__dirname, 'data.json');
+mongoose.connect(config.mongodb.connectionUrl, { useNewUrlParser: true });
```
Comment (slowpokex) [inline src/server/server.js]: connect method returns promise, please make some notifications in then and catch methods

File: src/server/router.js [hunk 7]
```diff
@@ -0,0 +1,15 @@
+const router = require('express').Router();
+
+router.use('/',
+    require('./models/user/router'),
```
Comment (slowpokex) [inline src/server/router.js]: You have used this require twice, can you extract into variable?

File: src/server/models/user/router.js [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+            .then((user) => {
+                if(user) {
+                    throw new Error('User with this login already exists');
+                }
+                return new User(req.body).save();
+            })
+            .then((user) => {
+                req.login(user, (err) => {
+                    if(err) {
+                        throw new Error('Auth Error');
```
Comment (slowpokex) [inline src/server/models/user/router.js:57]: Be careful with this conditions

File: src/server/models/news/router.js [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+                res.json(data)
+            })
+    })
+    .delete((req, res, next) => {
+        News.remove({ _id: req.params.id })
+            .then((data) => res.json(data))
+            .catch(next);
+    });
+
+router.route('/api/my-news')
```
Comment (slowpokex) [inline src/server/models/news/router.js:25]: post = for adding new news
get = get all news

Comment (slowpokex) [reply src/server/models/news/router.js:25]: put here isn't correct

File: src/server/models/news/router.js [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+const fs = require('fs');
+const util = require('util');
+const News = require('./model');
+
+const readFileAsync = util.promisify(fs.readFile);
+const writeFileAsync = util.promisify(fs.writeFile);
+
+const storageFilePath = path.resolve(__dirname, '../data.json');
+
+router.route('/api/my-news/:id?')
```
Comment (slowpokex) [inline src/server/models/news/router.js]: get = get particular news
put = update particular news
delete = delete particular news

Comment (slowpokex) [reply src/server/models/news/router.js]: you forgot about put for updating concrete news

File: src/server/models/news/router.js [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+    .get((req, res, next) => {
+        if (req.params.id !== undefined) {
+            News.findById(req.params.id)
```
Comment (slowpokex) [inline src/server/models/news/router.js]: Can you use facade to isolate schemas in controllers??

Comment (slowpokex) [reply src/server/models/news/router.js]: Straight using schemas in controllers aren't good

File: src/constants.js [hunk 12]
```diff
@@ -15,5 +15,19 @@ export const CONSTANTS = {
...
+    get LOGIN_URL() {
+        return `${this.BASE_URL}/api/login`;
+    },
+    get LOGOUT_URL() {
+        return `${this.BASE_URL}/api/logout`;
+    },
+    get REGISTRATION_URL() {
+        return `${this.BASE_URL}/api/registration`;
+    },
+    HTTP_STATUSES: {
```
Comment (slowpokex) [inline src/constants.js:28]: npm i -S http-status
https://www.npmjs.com/package/http-status

**Review by slowpokex — CHANGES_REQUESTED** [review]
Please make some changes if you have enough time for it

File: src/server/models/user/router.js (hunk 8 above)
Comment (slowpokex) [reply src/server/models/user/router.js:57]: You don't need to fix it - it's okay

File: src/app/AppView.js (hunk 1 above)
Comment (dzmitrybek) [reply src/app/AppView.js]: fixed

File: src/app/AppView.js (hunk 2 above)
Comment (dzmitrybek) [reply src/app/AppView.js]: fixed

File: src/server/models/news/model.js (hunk 3 above)
Comment (dzmitrybek) [reply src/server/models/news/model.js]: fixed

File: src/server/models/user/model.js (hunk 4 above)
Comment (dzmitrybek) [reply src/server/models/user/model.js]: fixed

File: src/server/models/user/router.js (hunk 5 above)
Comment (dzmitrybek) [reply src/server/models/user/router.js]: fixed

File: src/server/models/news/router.js (hunk 9 above)
Comment (dzmitrybek) [reply src/server/models/news/router.js:25]: fixed

File: src/server/models/news/router.js (hunk 10 above)
Comment (dzmitrybek) [reply src/server/models/news/router.js]: fixed

File: src/server/models/news/router.js (hunk 11 above)
Comment (dzmitrybek) [reply src/server/models/news/router.js]: fixed

**Pull request merged**